eclipse-scout-core/src/table/Table.ts

```diff
@@ -5713,15 +5705,172 @@ export class Table extends Widget implements TableModel, Filterable<TableRow> {
     return this.visibleColumns().some(column => column instanceof NumberColumn && column.aggregationFunction !== 'none');
   }
 
+  setColumns(columnOrModels: ObjectOrChildModel<Column<any>>[], initValue?: any | ((Column, TableRow) => any)) {
+    if (arrays.equals(this.columns, columnOrModels)) {
+      return;
+    }
+    this._setColumns(columnOrModels, initValue);
+    if (this.rendered) {
+      this._renderColumns();
+    }
+  }
+
+  protected _setColumns(columnOrModels: ObjectOrChildModel<Column<any>>[], initValue?: any | ((Column, TableRow) => any)) {
+    // ensure initValue function
+    if (initValue === undefined) {
+      initValue = null;
+    }
+    if (!objects.isFunction(initValue)) {
+      const value = initValue;
+      initValue = (column: Column, row: TableRow) => value;
+    }
+
+    // ensure columns
+    const columns = columnOrModels
+      .map(column => this._ensureColumn(column))
+      .filter(column => !!column);
+
+    const columnsOld = [...this.columns];
```

Comment:
I personally prefer the human readable writing -> oldColumns instead of columnsOld